source/usage-examples/watch.txt

```diff
@@ -0,0 +1,171 @@
+=================
+Watch for Changes
+=================
+
+.. default-domain:: mongodb
+
+You can keep track of changes to data in MongoDB, such as changes to a
+collection, database, or deployment, by opening a **change stream**. A change
+stream allows applications to watch for changes to data and react to them.
+The change stream returns **change event** documents when changes occur.
+You can open a change stream by calling the ``watch()`` method on
+a ``MongoCollection``, ``MongoDatabase``, or ``MongoClient`` object:
+
+.. code-block:: java
+
+   ChangeStreamIterable<Document> changeStream = database.watch();
+
+The ``watch()`` method optionally takes an **aggregation pipeline**  which
+consists of an array of **stages** as the first parameter to filter and
+transform the change event output as follows:. 
+
+.. code-block:: java
+
+   List<Bson> pipeline = Arrays.asList(
+                           Aggregates.match(
+                              Filters.lt("fullDocument.runtime", 15)));
+   ChangeStreamIterable<Document> changeStream = database.watch(pipeline);
+
+The ``watch()`` method returns an instance of ``ChangeStreamIterable``, a class
+that offers several methods to access, organize, and traverse the results.
+``ChangeStreamIterable`` also inherits methods from its parent class,
+``MongoIterable`` which implements the core Java interface ``Iterable``.
+You can call ``forEach()`` on the ``ChangeStreamIterable`` to handle
+events as they occur, or you can use the ``iterator()`` method which
+returns a ``MongoCursor`` instance that you can use to traverse the results.
+You can call methods on the ``MongoCursor`` such as ``hasNext()`` to check
+whether additional results exist, or ``next()`` to return the next document
+in the collection. Unlike the ``MongoCursor`` returned by other queries,
+a ``MongoCursor`` associated with a change stream will wait until a
+change event arrives before returning a result from ``next()`` and
+``hasNext()`` instead of throwing a ``java.util.NoSuchElementException``.
```

Comment:
Could we either separate the description of the exception in a new sentence (hasNext() will never throw that exception)? And if there are more events pending that you have not consumed with `next()` from the cursor, it will not throw it either. 
I think if we're going to describe the exception case, it should be separated and very clear in what circumstance it may happen.